From 554d15ecb5e56931a98a0516a7e000eee057111a Mon Sep 17 00:00:00 2001
From: Primoz Fiser <primoz.fiser@norik.com>
Date: Sat, 24 Jan 2026 08:25:04 -0500
Subject: [PATCH] Disable caldata prefetch for SDIO bus

- when enabled on iMX6 platform we get:
	mmc0: Timeout waiting for hardware interrupt.

if we try to disable wireless interface with:
	ifconfig wlan0 down

Afterwards, the wireless interface & SDIO bus are left
unresponsive and cannot be easily brought back to operation.

Signed-off-by: Primoz Fiser <primoz.fiser@norik.com>
---
 drivers/net/wireless/ath/ath10k/debug.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/wireless/ath/ath10k/debug.c b/drivers/net/wireless/ath/ath10k/debug.c
index e45ea59e3e42..d6c0ac0322bd 100644
--- a/drivers/net/wireless/ath/ath10k/debug.c
+++ b/drivers/net/wireless/ath/ath10k/debug.c
@@ -1264,6 +1264,9 @@ static int ath10k_debug_cal_data_fetch(struct ath10k *ar)
 	if (ar->hw_params.cal_data_len == 0)
 		return -EOPNOTSUPP;
 
+	if (ar->hif.bus == ATH10K_BUS_SDIO)
+		return -EINVAL;
+
 	hi_addr = host_interest_item_address(HI_ITEM(hi_board_data));
 
 	ret = ath10k_hif_diag_read(ar, hi_addr, &addr, sizeof(addr));
-- 
2.51.0

